include/mvdtool/mvd_generic.hpp

```diff
@@ -64,16 +64,16 @@ inline std::shared_ptr<File> open(const std::string& filename,
     case MVDType::MVD2:
         throw std::runtime_error("MVD2 not supported in the new MVD API");
     case MVDType::MVD3:
+        mvdfile.reset(new MVD3::MVD3File(filename));
         try {
-            mvdfile.reset(new MVD3::MVD3File(filename));
             mvdfile->size();
         } catch (...) {
             throw std::runtime_error("invalid MVD3 file");
         }
         break;
     default:
+        mvdfile.reset(new SonataFile(filename, population));
         try {
-            mvdfile.reset(new SonataFile(filename, population));
             mvdfile->size();
```

Comment:
This will never throw, it just accesses the `size_` member…

Reply:
Matthias is right. Maybe instead of just throwing a `std::runtime_error` we could catch the exception from `libsonata` and rethrow it?

Reply:
Good catch, Sonata is different in that respect. The idea of avoiding catch and rethrowing is to avoid losing the original message, which very often is more informative than a plain "Invalid File"